src/zarr/core/metadata/v2.py

```diff
@@ -62,7 +62,10 @@ def __init__(
         order_parsed = parse_indexing_order(order)
         dimension_separator_parsed = parse_separator(dimension_separator)
         filters_parsed = parse_filters(filters)
-        fill_value_parsed = parse_fill_value(fill_value, dtype=data_type_parsed)
+        if fill_value is None:
+            fill_value_parsed = None
+        else:
+            fill_value_parsed = parse_fill_value(fill_value, dtype=data_type_parsed)
```

Comment:
rather than putting this check here, can we instead allow `parse_fill_value` to return `None`?

Reply:
That already worked.